### DISCUSSION TITLE
Feat: add support for checking if there's any complete item selected

### DISCUSSION THREAD

**Discussion by JS-Zheng** [neoclide/coc.nvim#3972] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I always use the following snippet to check if there's any complete item be selected (e.g., #262),
However, this feature has been removed in commit [7a50d4](https://github.com/neoclide/coc.nvim/commit/7a50d4d9513362aa21cf42646ceca0cb41cb0dbc).
```vim
coc#rpc#request('hasSelected', [])
```

Many of my coc scripts need this feature, for example,
I like to confirm the completion only when I explicitly select a complete item and then press `<CR>`:
```vim
inoremap <silent><expr> <CR> pumvisible() && coc#rpc#request('hasSelected', [])
                              \? coc#_select_confirm()
                              \: "\<C-g>u\<CR>\<c-r>=coc#on_enter()\<CR>"
```

**Describe the solution you'd like**
Provide a `coc#is_selected()` or `coc#_selected()` API
or just restore `hasSelected` action.

**Comment by chemzqm** [comment, marked as answer]
You can use `coc#pum#visible() && coc#pum#info()['index'] != -1`
